apps/sms/views/inbox/js/inbox.js

```diff
@@ -806,29 +708,16 @@ var InboxView = {
     Threads.delete(threadId);
 
     // Cleanup the DOM
-    this.removeConversationDOM(threadId);
+    var index = this.listData.findIndex((data) => data.id === threadId);
+    if (index > -1) {
+      this.listData.splice(index, 1);
+      this.list.setModel(this.listData);
```

Comment:
If we want a more clever fast list, we should just the library instead of the component: https://github.com/fxos-components/fastlist